src/containers/LearningResourcePage/components/LearningResourceTaxonomy.jsx

```diff
@@ -9,133 +9,466 @@
-        relevances: [],
       },
     };
+    this.retriveBreadCrumbs = this.retriveBreadCrumbs.bind(this);
+    this.stageTaxonomyChanges = this.stageTaxonomyChanges.bind(this);
+    this.removeConnection = this.removeConnection.bind(this);
+    this.setPrimaryConnection = this.setPrimaryConnection.bind(this);
+    this.getSubjectTopics = this.getSubjectTopics.bind(this);
+    this.onChangeSelectedResource = this.onChangeSelectedResource.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.updateSubject = this.updateSubject.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.fetchTaxonomy = this.fetchTaxonomy.bind(this);
+    this.fetchTaxonomyChoices = this.fetchTaxonomyChoices.bind(this);
+    this.getResourceId = this.getResourceId.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchTaxonomy();
+    this.fetchTaxonomyChoices();
   }
 
-  async componentDidMount() {
-    const { model } = this.props;
+  onChangeSelectedResource(e) {
+    const {
+      taxonomy: { availableResourceTypes },
+    } = this.state;
+    const options = e.target.value.split(',');
+    const selectedResource = availableResourceTypes.find(
+      resourceType => resourceType.id === options[0],
+    );
+    const val = [
+      {
+        name: selectedResource.name,
+        id: selectedResource.id,
+      },
+    ];
+    if (options.length > 1) {
+      const subType = selectedResource.subtypes.find(
+        subtype => subtype.id === options[1],
+      );
+      val.push({
+        id: subType.id,
+        name: subType.name,
+        parentId: selectedResource.id,
+      });
+    }
+
+    this.stageTaxonomyChanges({
+      resourceTypes: val,
+    });
+  }
 
+  async getSubjectTopics(subjectid) {
+    if (
+      this.state.structure.some(
+        subject => subject.id === subjectid && subject.topics,
+      )
+    ) {
+      return;
+    }
     try {
-      const [resourceTypes, filters, topics, relevances] = await Promise.all([
-        fetchResourceTypes(model.language),
-        fetchFilters(model.language),
-        fetchTopics(model.language),
-        fetchRelevances(model.language),
+      this.updateSubject(subjectid, { loading: true });
+      const allTopics = await fetchSubjectTopics(subjectid);
+      const groupedTopics = groupTopics(allTopics);
+      this.updateSubject(subjectid, { loading: false, topics: groupedTopics });
+    } catch (e) {
+      handleError(e);
+    }
+  }
+
+  setPrimaryConnection(id) {
+    const { topics } = this.state.taxonomyChanges;
+
+    this.stageTaxonomyChanges({
+      topics: topics.map(topic => ({
+        ...topic,
+        primary: topic.id === id,
+      })),
+    });
+  }
+
+  async getResourceId() {
+    const { articleId, language } = this.props;
+    let resourceId = '';
+    const resource = await queryResources(articleId, language);
+    if (resource.length > 0) resourceId = resource[0].id;
+    return resourceId;
+  }
+
+  async fetchTaxonomy() {
```

Comment:
Maybe move all fetching logic to taxonomyApi? (i.e. export fetchTaxonomy and fetchTaxonomyChoices)